src/components/store/ui/Product.tsx

```diff
@@ -0,0 +1,212 @@
+    </ProductPrimitive.Price>
+  );
+});
+
+ProductPrice.displayName = 'ProductPrice';
+
+export const ProductCompareAtPrice = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.CompareAtPrice>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.CompareAtPrice>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.CompareAtPrice
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-lg text-content-muted line-through hidden data-[discounted]:inline',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.CompareAtPrice>
+  );
+});
+
+ProductCompareAtPrice.displayName = 'ProductCompareAtPrice';
+
+export const ProductSlug = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Slug>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Slug>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Slug
+      {...props}
+      ref={ref}
+      className={cn(
+        'text-content-secondary font-mono text-sm',
+        props.className
+      )}
+    >
+      {props.children}
+    </ProductPrimitive.Slug>
+  );
+});
+
+ProductSlug.displayName = 'ProductSlug';
+
+export const ProductRaw = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Raw>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Raw>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Raw
+      {...props}
+      ref={ref}
+      className={cn('block', props.className)}
+    >
+      {props.children}
+    </ProductPrimitive.Raw>
+  );
+});
+
+ProductRaw.displayName = 'ProductRaw';
+
+export const ProductMediaGallery = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.ProductMediaGallery>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.ProductMediaGallery>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.ProductMediaGallery {...props} ref={ref}>
+      {props.children}
+    </ProductPrimitive.ProductMediaGallery>
+  );
+});
+
+ProductMediaGallery.displayName = 'ProductMediaGallery';
+
+export const ProductVariants = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Variants>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Variants>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Variants
+      {...props}
+      ref={ref}
+      className={cn('space-y-4', props.className)}
+    >
+      {props.children}
+    </ProductPrimitive.Variants>
+  );
+});
+
+ProductVariants.displayName = 'ProductVariants';
+
+export const ProductModifiers = React.forwardRef<
+  React.ElementRef<typeof ProductPrimitive.Modifiers>,
+  React.ComponentPropsWithoutRef<typeof ProductPrimitive.Modifiers>
+>((props, ref) => {
+  return (
+    <ProductPrimitive.Modifiers
+      {...props}
+      ref={ref}
+      className={cn('space-y-4', props.className)}
+    >
+      {props.children}
+    </ProductPrimitive.Modifiers>
+  );
+});
+
+ProductModifiers.displayName = 'ProductModifiers';
+
+export const ProductVariantOptions = React.forwardRef<
```

Comment:
Why did you have to redefine it and not just 
`export const ProductVariantOptions = ProductPrimitive.ProductVariantOptions`
?